src/test/java/org/folio/circulation/support/utils/DateTimeUtilTests.java

```diff
@@ -0,0 +1,461 @@
+package org.folio.circulation.support.utils;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class DateTimeUtilsTests {
+
+  @ParameterizedTest
+  @MethodSource("shouldGetCorrectStartOfYearForZonedDateTimeParameters")
+  void shouldGetCorrectStartOfYearForZonedDateTime(int id, ZonedDateTime from, ZonedDateTime expected, ZoneId zone) {
+    final ZonedDateTime result = zone == null
+      ? DateTimeUtil.atStartOfYear(from)
+      : DateTimeUtil.atStartOfYear(from, zone);
```

Comment:
I would prefer there weren't branching statements in tests, I find it harder to follow what is meant to be happening

Reply:
It greatly simplified how much testing I needed to perform.
Otherwise I would have to write 2x the test functions in this file.

Tests like `DateTimeUtil.atStartOfYear(XXX)` are wrappers to functions like `DateTimeUtil.atStartOfYear(XXX, YYY)` and so are really the same test but with a NULL for YYY hidden for convenience for the first method.

Reply:
> Otherwise I would have to write 2x the test functions in this file.

Personally, I would prefer double the number of simpler tests.

Any way, that's likely only my preference.

> Tests like `DateTimeUtil.atStartOfYear(XXX)` are wrappers to functions like `DateTimeUtil.atStartOfYear(XXX, YYY)` and so are really the same test but with a NULL for YYY hidden for convenience for the first method.

Why would there be a `startOfYear` method that didn't take year as a parameter?

Reply:
```
Why would there be a startOfYear method that didn't take year as a parameter?
```
These functions find the start of some date based on the given date, or date and time, parameter with optional time zone parameter.

Such as `(ZonedDateTime, ZoneId)` or `(DateTime, DateTimeZone)` where the time zone can be null.

Reply:
Ok, I was confused by the use of `YYY` as an example parameter, apologies.

I get the sense that my suggestion isn't going to change your stance on these tests.